File: src/main/java/tk/skulk/plugin/core/extensions/foodbehaviour/FoodBehaviourExtension.java
package tk.skulk.plugin.core.extensions.foodbehaviour;

import org.jetbrains.annotations.NotNull;
import tk.skulk.plugin.core.Plugin;
import tk.skulk.plugin.core.extensions.foodbehaviour.listeners.CookieResetInsomniaListener;
import tk.skulk.plugin.wrapper.BaseExtension;

public final class FoodBehaviourExtension extends BaseExtension {
    public FoodBehaviourExtension(final @NotNull Plugin plugin) {
        super(plugin);
    }

    @Override
    protected void initListeners() {
        new CookieResetInsomniaListener(this);
    }
}

File: src/main/kotlin/tk/skulk/plugin/core/extensions/foodBehaviour/FoodBehaviourExtension.kt
package tk.skulk.plugin.core.extensions.foodBehaviour

import tk.skulk.plugin.core.SNPlugin
import tk.skulk.plugin.core.extensions.foodBehaviour.listeners.CookieResetInsomniaListener
import tk.skulk.plugin.framework.Extension

class FoodBehaviourExtension(plugin: SNPlugin) : Extension<SNPlugin>(plugin) {
    override val listeners = listOf {
        CookieResetInsomniaListener(this)
    }
}
